js/battle.js

```diff
@@ -524,18 +524,21 @@ var Pokemon = (function () {
 			this.baseAbility = ability;
 		}
 	};
+	Pokemon.prototype.htmlName = function () {
+		return '<span class="pokemon" title="' + this.species + '">' + (this.side.battle.ignoreOpponent || this.side.battle.ignoreNicks ? this.species : Tools.escapeHTML(this.name)) + '</span>';
```

Comment:
I don't like `class="pokemon"`, you're playing with namespace fire. Something like `battle-nickname` and `battle-nickname-foe`.